projects/angular-grid-layout/src/lib/grid.component.ts

```diff
@@ -40,23 +40,27 @@ function getDragResizeEventData(gridItem: KtdGridItemComponent, layout: KtdGridL
 
 
 function layoutToRenderItems(config: KtdGridCfg, width: number, height: number): KtdDictionary<KtdGridItemRenderData<number>> {
-    const {cols, rowHeight, layout} = config;
-
+    const {cols, rowHeight, layout, gap} = config;
+    const widthExcludinggap = width - Math.max((gap * (cols - 1)), 0);
+    const itemWidthPerColumn = (widthExcludinggap / cols);
     const renderItems: KtdDictionary<KtdGridItemRenderData<number>> = {};
     for (const item of layout) {
         renderItems[item.id] = {
             id: item.id,
-            top: item.y === 0 ? 0 : item.y * rowHeight,
-            left: item.x * (width / cols),
-            width: item.w * (width / cols),
-            height: item.h * rowHeight
+            top: item.y === 0 ? 0 : item.y * rowHeight + gap * item.y,
+            left:
+              item.x === 0 || item.x === cols - 1
+                  ? item.x * itemWidthPerColumn
+                  : item.x * itemWidthPerColumn + gap * item.x,
```

Comment:
I found a bug here, happens when I try to move an item to the last column. It does not get well calculated. Why not just set `left: item.x * itemWidthPerColumn + gap * item.x`? It seems that would work in all cases, is there some edge case I am missing?

Reply:
Ah yes their is an issue here. The edge case I was trying to address is the first and last column only have a gap to its right and left respectively. Since we are assigning the left property here, the check against the last column is not needed because it will still needs to move over the additional gap spacing, but the first column check needs to remain so items in the first column have a translateX of 0 instead of whatever the gap is set to.